UnityProject/Assets/Scripts/Items/Weapons/Gun.cs

```diff
@@ -281,6 +287,7 @@ public virtual bool WillInteract(HandActivate interaction, NetworkSide side)
 			//try ejecting the mag if external
 			if (CurrentMagazine != null && allowMagazineRemoval && !MagInternal)
 			{
+				UnloadMagSound();
```

Comment:
this method, ``WillInteract`` runs once in the client requesting the interaction and then on server to ensure it is a valid interaction. Did you test this change in client & server environment? Does it play the sound twice?

It is also not the best place to call the sound, imo. It would be better to call it inside the method that handles unloading and loading mags respectively.